sbat.c

```diff
@@ -335,4 +335,62 @@ parse_sbat_var(list_t *entries)
 	FreePool(data);
 	return EFI_INVALID_PARAMETER;
 }
+
+EFI_STATUS
+init_sbat()
+{
+	EFI_STATUS efi_status = EFI_SUCCESS;
+	UINT32 attributes = 0;
+
+	CHAR8 *sbat;
+	UINTN sbatsize = 4096;
+
+	sbat = AllocateZeroPool(sbatsize);
+
+	/* read and get check attributes */
+	efi_status = gRT->GetVariable(L"SBAT", &SHIM_LOCK_GUID, &attributes,
+			&sbatsize, sbat);
+
```

Comment:
I need to check of the `GetVariable()` EFI service will behave here but I wonder if we shouldn't first do a call with `sbat == NULL` to get the size and allocate the buffer if the return value is `EFI_BUFFER_TOO_SMALL` or something like that.

Another option is to use one of the variable API from `gnu-efi`, for example `LibGetVariable()` or `LibGetVariableAndSize()`.

Reply:
I confirmed that calling with GetVariable() 0 and NULL returns the size. Fixing.